File: src/Bk.APG.Business/Models/WorklistFilterParameters.cs
namespace Bk.APG.Business.Models;

public class WorklistFilterParameters
{
    public string? Committee { get; set; }
    public IEnumerable<Guid>? DepartmentIds { get; set; }
    public IEnumerable<Guid>? OfficeIds { get; set; }
    public IEnumerable<Guid>? WorklistTaskStateIds { get; set; }
    public IEnumerable<Guid>? WorklistTaskTypeIds { get; set; }
    public string? AssignedBy { get; set; }
    public string? AssignedTo { get; set; }
    public DateTime? CreatedFrom { get; set; }
    public DateTime? CreatedTo { get; set; }
    public DateOnly? DueDateFrom { get; set; }
    public DateOnly? DueDateTo { get; set; }
}

File: src/Bk.APG.UI/src/api/WorklistFilterParameters.ts
import {CommitteeDetails} from './CommitteeDetails';
import {Department} from './Department';
import {Office} from './Office';
import {WorklistTaskState} from './WorklistTaskState';
import {WorklistTaskType} from './WorklistTaskType';

export interface WorklistFilterParameters {
    committee?: CommitteeDetails['description'];
    departments?: Department['id'][];
    offices?: Office['id'][];
    worklistTaskTypes?: WorklistTaskType['id'][];
    worklistTaskStates?: WorklistTaskState['id'][];
    assignedBy?: string;
    assignedTo?: string;
    createdFrom?: Date;
    createdTo?: Date;
    dueDateFrom?: Date;
    dueDateTo?: Date;
}
